src/store/trails/actions.js

```diff
@@ -585,6 +597,40 @@ export const editTreasury = async function (
   return notification.status === 'success';
 };
 
+export const editTreasurySettings = async function (
+  { commit, dispatch},
+  { settings, treasury }
+) {
+  const notification = {
+    icon: 'fas fa-edit',
+    title: 'notifications.trails.editTreasury',
+    content: `Treasury: ${treasury.title}`,
+  };
+  try {
+    let actions = settings.filter((x,i) => treasury.settings[i].value != x.value)
+    .map(x => ({
+      account: 'telos.decide',
+      name: 'toggle',
+      data: {
+        treasury_symbol: supplyToAsset(treasury.symbol),
+        setting_name: x.key
+      },
+    }));
+
+    const transaction = await this.$api.signTransaction(actions);
+    commit('updateTreasurySettings', { settings, treasury });
+    await dispatch('fetchTreasury', supplyToSymbol(treasury.max_supply));
+
+    notification.status = 'success';
```

Comment:
What do you think about we move away from `'success'` and `'error'` and use an Enum instead?

Reply:
This is something that is all over the store folder and is not a minor fix. I propose to leave this fix out of the current PR and create a separate PR, especially for this.

I created this issue https://github.com/telosnetwork/app-telos-native/issues/225 for that but it needs grooming. Could you please edit and develop the body of that issue? (change the title also if you wish)